src/hash.rs

```diff
@@ -0,0 +1,85 @@
+use rustls::crypto::hash;
+use rust_symcrypt::hash::{Sha256State, Sha384State, HashState, sha256, sha384};
+
+
+/// Impl rustls::hash::Hash trait for Sha256, Underlying code can be found in rust_symcrypt::hash
+pub struct Sha256;
+pub struct Sha256Context(Sha256State);
+
+impl hash::Hash for Sha256 {
+    fn algorithm(&self) -> hash::HashAlgorithm {
+        hash::HashAlgorithm::SHA256
+    }
+
+    fn output_len(&self) -> usize {
+        32
+    }
+
+    fn start(&self) -> Box<dyn hash::Context> {
+        Box::new(Sha256Context(Sha256State::new()))
+    }
+
+    fn hash(&self, data: &[u8]) -> hash::Output {

```

Comment:
>hash [](http://example.com/codeflow?start=7&length=4)

This should be compute

Reply:
from rustls code:

`/// Return the output of this hash function with input `data`.
    fn hash(&self, data: &[u8]) -> Output;`

Reply:
Did they change the name of this function in the hash::Hash trait:

    /// Return the output of this hash function with input `data`.
    fn compute(&self, data: &[u8]) -> Output;
}

Before it was compute and not hash

Reply:
I see that they changed the function name to hash.